radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/network/FProposalsPerViewDropper.java

```diff
@@ -59,12 +59,17 @@ public boolean test(MessageInTransit msg) {
 			final View view = proposal.getVertex().getView();
 			final Set<BFTNode> nodesToDrop = proposalToDrop.computeIfAbsent(view, v -> {
 				List<BFTNode> nodes = new LinkedList<>(validatorSet);
-				Set<BFTNode> nextFaultySet = new HashSet<>();
-				for (int i = 0; i < faultySize; i++) {
-					BFTNode nextFaultyNode = nodes.remove(random == null ? i : random.nextInt(nodes.size()));
```

Comment:
Slightly nasty bug there.  Why did this not trigger some sort of out of range exception?

I guess we are using `faultySize <<< validatorSet.size()`.